### PR TITLE
Guard muelu tests

### LINKED ISSUE #657: Need a way to run MueLu tests only if TrilinosVersion >= 10.12
The MueLu tests currently run whenever you have Trilinos installed (starting with #214 ), but they should only run if TRILINOS_VERSION >= 10.12.

@tamiko : Any good way of doing this?

### PR DESCRIPTION
**Pull request by tamiko**
This pull request consists of a series of patches to guard muelu tests such
that they are only run if Trilinos 11.14 is available (11.14 to be
consistent with our configuration).

Closes #657.

a6ea163 (Matthias Maier, 2 minutes ago)
   Documentation: Update testsuite documentation

e0d50e6 (Matthias Maier, 14 minutes ago)
   Testsuite: Guard trilinos tests using muelu

e974396 (Matthias Maier, 20 minutes ago)
   Testsuite: Augment feature constraints with version number

   Beside the already introduced constraints

     with_feature=(yes|true|on)
    with_feature=(no|false|off)

   it is now also possible to specify a version number:

     with_feature=x.y.z

   A test with this constraint will be picked up if

    a) deal.II is configured with "feature"

    b) DEAL_II_FEATURE_VERSION is defined (in deal.IIConfig.cmake) and
      DEAL_II_FEATURE_VERSION is "version greater or equal" than "x.y.z"

   A version number "x.y.z" must satisfy the regular expression
   [0-9]+(\.[0-9]+)*

c082145 (Matthias Maier, 84 minutes ago)
   CMake: Also export feature versions in deal.IIConfig.cmake and
   Make.global_options

   With this commit the version string of enabled features (with version
   information) is also exported to deal.IIConfig.cmake and
   Make.global_options

### CODE REVIEW

File: cmake/config/CMakeLists.txt [hunk 1]
```diff
@@ -205,7 +204,7 @@ SET(_files
   )
 SET(_makefiles
   ${CMAKE_BINARY_DIR}/${DEAL_II_SHARE_RELDIR}/Make.global_options
-  ${CMAKE_CURRENT_BINARY_DIR}/${DEAL_II_SHARE_RELDIR}/Make.global_options
+  ${CMAKE_CURRENT_BINARY_DIR}/Make.global_options
```
Comment (tjhei) [inline cmake/config/CMakeLists.txt]: did the location of Make.global_options change?

**Comment by tjhei** [comment]
looks good otherwise and tested that it works here.

File: cmake/config/CMakeLists.txt (hunk 1 above)
Comment (tamiko) [inline cmake/config/CMakeLists.txt]: No, this line was wrong.

I got sidetracked a bit and forgot to put this change in a separate commit.
